.github/workflows/_pypi.yml

```diff
@@ -0,0 +1,17 @@
+on:
+  workflow_call:
+
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    environment: release
+
+    steps:
+      - name: Download dist artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: dist
+          path: dist
+
+      - name: Publish to PyPI using trusted publishing
+        uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
@mrakitin I'm not sure if this will allow the current tagging paradigm (`v0.1.2` instead of `0.1.2`). Is this something we care about? If so I can look into preserving it.

Reply:
@callumforrester, I think it's handled properly. I used this action in our [`hextools` package](https://github.com/NSLS-II-HEX/hextools/blob/00d8456c1e3bf0d278940d1f60e6a182d9e27655/.github/workflows/cd.yml#L47-L48). The release on GH has `v` in the version, while on PyPI it's published w/o it:
- https://github.com/NSLS-II-HEX/hextools/releases/tag/v0.1.3
- https://pypi.org/project/hextools/0.1.3/